File: app/src/main/java/com/frank/ffmpeg/util/ThreadPoolUtil.java
package com.frank.ffmpeg.util;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class ThreadPoolUtil {

    public static ExecutorService executeSingleThreadPool(Runnable runnable) {
        ExecutorService executor = Executors.newSingleThreadExecutor();
        executor.submit(runnable);
        return executor;
    }

}

File: app/src/main/java/com/frank/ffmpeg/util/ThreadPoolUtil.kt
package com.frank.ffmpeg.util

import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

object ThreadPoolUtil {

    private val executor = Executors.newSingleThreadExecutor()

    fun executeSingleThreadPool(runnable: Runnable): ExecutorService {
        executor.submit(runnable)
        return executor
    }

}
